syncing/src/test/kotlin/maru/syncing/beaconchain/pipeline/DownloadBlocksTest.kt

```diff
@@ -245,6 +245,70 @@ class DownloadBlocksTest {
     }
   }
 
+  @Test
+  fun `always select a random peer if there are available peers when purely random selection is true`() {
```

Comment:
I suggest to test it differently. Have a list of 1 peer, with `randomStatus(0U)` and the test would check if it actually returns the only available option. The filtering implementation would fail in this case